project/webapp/controller/ProductsList.controller.js

```diff
@@ -1,15 +1,193 @@
+      TABLE_MODEL_NAME: "tableModel",
+      FILTER_BAR_MODEL_NAME: "filterBarModel",
+      TOKEN_REMOVED_TYPE: "removed",
+
+      onInit() {
+        productModel.initModel();
+        filterBarModel.initFilterBarModel();
+        const oModel = productModel.getModel();
+        this.oFilterBarModel = filterBarModel.getFilterBarModel();
+
+        this.oTableModel = new JSONModel({
+          isProductsSelected: false,
+        });
+
+        this.getView().setModel(oModel, this.APP_MODEL_NAME);
+        this.getView().setModel(this.oTableModel, this.TABLE_MODEL_NAME);
+        this.getView().setModel(this.oFilterBarModel, this.FILTER_BAR_MODEL_NAME);
+      },
+
+      onSelectProduct(bProductSelected) {
+        this.oTableModel.setProperty("/isProductsSelected", bProductSelected);
+      },
+
+      onSearchProducts(oEvent) {
+        const oTableBinding = this.byId("productList").getBinding("items");
+        const aFilters = this._getAllFilters(oEvent);
+
+        oTableBinding.filter(aFilters);
+      },
+
+      onClearFilters() {
+        const oTableBinding = this.byId("productList").getBinding("items");
+
+        this.byId("releaseDate").setValue(null);
+        this.byId("supplier").setTokens([]);
+        this.byId("searchName").setValue(null);
+        this.byId("categorySelect").setSelectedKeys([]);
+
+        oTableBinding.filter(null);
+      },
+
+      _getSearchNameFilter() {
+        const sSearchQuery = this.byId("searchName").getProperty("value");
+
+        return sSearchQuery.length ? new Filter("name", FilterOperator.Contains, sSearchQuery) : null;
+      },
+
+      _getCategoriesFilter() {
+        const aSelectedCategories = this.byId("categorySelect").getProperty("selectedKeys");
+        const aFilters = [];
+
+        if (!!aSelectedCategories.length) {
+          aSelectedCategories.map((sSelectedKey) => {
+            aFilters.push(
+              new Filter({
+                path: "categories",
+                operator: FilterOperator.EQ,
+                value1: sSelectedKey,
+                test: (aCategories) => {
+                  const aResult = aCategories.filter(({id}) => id === sSelectedKey);
+
+                  return !!aResult.length;
+                },
+              })
+            );
+          });
+        }
+
+        return aFilters.length ? aFilters : null;
+      },
+
+      _getDateFilter() {
+        const sDate = this.byId("releaseDate");
+        const sDateStart = sDate.getDateValue();
+        const sDateEnd = sDate.getSecondDateValue();
+
+        if (!sDateStart || !sDateEnd) {
+          return null;
+        }
+
+        return new Filter({
+          path: "releaseDate",
+          operator: FilterOperator.BT,
+          value1: sDateStart.toISOString(),
+          value2: sDateEnd.toISOString(),
         });
+      },
+
+      _getCurrTokens(oEvent) {
+        let aSuppliersTokens = this.byId("supplier").getTokens();
+        const oMultiInput = oEvent.getParameters();
```

Comment:
this is not best variable name, as oEvent.getParameters() will not return you Input
better use oEventParameters